### ISSUE TITLE
HTTP File Server Extended Testing - learnyounode #11 

### ISSUE THREAD

**Issue by mdavidjohnson** [nodeschool/discussions#797]
Here's my code:

```
var fs = require('fs');
var http = require('http');

var portNumber = process.argv[2];
var pathName = process.argv[3];

var server = http.createServer(function handleRequest(request, response) {
  response.writeHead(200, {'content-Type': 'text/plain'});
  fs.createReadStream(pathName).pipe(response);
});

server.listen(Number(portNumber));

console.log('Http File Server running at 192.168.10.8:' + portNumber);
```

This code runs and verifies properly.

But, to test it further, I created the following TestText.txt file:

```
TestText.txt

This is an additional text file for further testing of httpFileServer.js - 
it will be used with the command sequence:

		node httpFileServer.js 8888 TestText.txt
		
and then we'll open a browser and navigate to http://192.168.10.8:8888

since 192.168.10.8 is the network address of the CentOS 6.6 linux machine.
```

Then, I entered the above specified command sequence, opened another terminal and navigated to the above specified url in my elinks browser, and it performed perfectly. I use Putty on my Windows 7 machine as the ssh client to open terminals to the CentOS machine.

My CentOS machine at 192.168.10.8 is physically networked to my Windows 7 machine at 192.168.10.16 - but when I open my FireFox browser on my Windows 7 machine and navigate to http://192.168.10.8:8888 the request times out. I've also tried it on port 80 with the same results.

The two machines ping each other with no difficulties, and I use WS_FTP to transfer files back and forth between the two machines, so I know the network connection is good.

My Windows 7 machine is setup as a proxy server (using the AnalogX Proxy) and I've added two aliases to my CentOS machine's /etc/bashrc file to turn the proxy on or off (in order to get out through my wireless router and "yum install", etc.). But the request from the Windows machine times out whether the proxy is on or off.

Would anyone have any suggestions concerning what I'm missing here?

**Comment by mdavidjohnson** [comment]
Hmmm...

I'm new here, so perhaps my request for assistance above was somehow improper?

I would have thought this to be a reasonable request, even though it somewhat exceeds the scope of the original challenge. An http server which only serves itself would seem to be rather narcissistic and incestuous. :-)

Could somebody at least point me in the right direction?

**Comment by mdavidjohnson** [comment]
BTW - How do I add the learnyounode label to this?

**Comment by mdavidjohnson** [comment]
Additional info:

I'm running node 0.10.35.

My iptables are set to ACCEPT for INPUT, FORWARD, and OUTPUT for the 192.168.10.8 network.

On the Windows 7 machine, the Windows firewall is turned off because Kaspersky Internet Security provides the firewall. That firewall is set to treat all the 192.168.10.x addresses as a (wide-open) trusted network. But, even with the Kaspersky firewall completely disabled, http://192.168.10.8:8888 still times out.

**Comment by mdavidjohnson** [comment]
@martinheidegger -

Thank you for adding the learnyounode label. Is that something the administrators have  to do, or is it something I can do when I initiate an issue?

**Comment by martinheidegger** [comment]
@mdavidjohnson Just to assure you: your request was not just proper, it was very clearly put. There are many requests in this issue board and its easy to loose track of one that is too difficult for a quick reply, I am sorry that you felt lost.

Unfortunately, I am still not in the position to help you. It seems to me like something in your network is blocking the ports (obviously). This could be software on either machine or any firewall that is in-between. Maybe you could try to run any other http server on that port and see if that is able to connect?

(This would surely ease diagnosing it)

Adding labels is only possible by people with permission, we will add that for you.

**Comment by mdavidjohnson** [comment]
I found the problem!

When I originally tried to set up iptables on the CentOS machine, I used the "iptables -A" command to APPEND the 192.168.10.x information, e.g.

```
     iptables -A INPUT -s 192.168.10.0/255.255.255.0 -j ACCEPT
     iptables -A FORWARD -s 192.168.10.0/255.255.255.0 -j ACCEPT
     iptables -A OUTPUT -s 192.168.10.0/255.255.255.0 -j ACCEPT
```

Unfortunately, that appended the info to the end of each list, after the existing REJECT line for that list (i.e. the INPUT, FORWARD, and OUTPUT lists). Once the system hits the REJECT line, it stops. Thus, even though I had appended the correct lines, those lines were never reached.

Instead, I should have used the "iptables -I" command, with no insert location number, to force my lines to the head of each list where they would be encountered first, before any other lines, e.g. 

```
     iptables -I INPUT -s 192.168.10.0/255.255.255.0 -j ACCEPT
     iptables -I FORWARD -s 192.168.10.0/255.255.255.0 -j ACCEPT
     iptables -I OUTPUT -s 192.168.10.0/255.255.255.0 -j ACCEPT
```

Once I did that, the problem was solved and I was able to reach the server on the CentOS machine with the http://192.168.10.8:8888 address in my Windows 7 machine's browser.
